src/Mobs/Path.cpp

```diff
@@ -388,23 +381,20 @@ void cPath::ProcessCell(cPathCell * a_Cell, cPathCell * a_Caller, int a_GDelta)
 cPathCell * cPath::GetCell(const Vector3i & a_Location)
 {
 	// Create the cell in the hash table if it's not already there.
-	cPathCell * Cell;
 	if (m_Map.count(a_Location) == 0)  // Case 1: Cell is not on any list. We've never checked this cell before.
 	{
-		Cell = new cPathCell();
-		Cell->m_Location = a_Location;
-		m_Map[a_Location] = UniquePtr<cPathCell>(Cell);
-		Cell->m_IsSolid = IsSolid(a_Location);
-		Cell->m_Status = eCellStatus::NOLIST;
+		m_Map[a_Location].m_Location = a_Location;
+		m_Map[a_Location].m_IsSolid = IsSolid(a_Location);
+		m_Map[a_Location].m_Status = eCellStatus::NOLIST;
 		#ifdef COMPILING_PATHFIND_DEBUGGER
 			#ifdef COMPILING_PATHFIND_DEBUGGER_MARK_UNCHECKED
 				si::setBlock(a_Location.x, a_Location.y, a_Location.z, debug_unchecked, Cell->m_IsSolid ? NORMAL : MINI);
 			#endif
 		#endif
-		return Cell;
+		return &m_Map[a_Location];
 	}
 	else
 	{
-		return m_Map[a_Location].get();
+		return &m_Map[a_Location];
```

Comment:
What does the else do here? you can just return  &m_Map[a_Location]; after closing the if